services/secondary-storage/controller/src/org/apache/cloudstack/secondarystorage/SecondaryStorageManagerImpl.java

```diff
@@ -521,6 +522,76 @@ public SecondaryStorageVmVO startNew(long dataCenterId, SecondaryStorageVm.Role
         return null;
     }
 
+    /**
+     * Get the default network for the secondary storage VM, based on the zone it is in. Delegates to
+     * either {@link #getDefaultNetworkForZone(DataCenter)} or {@link #getDefaultNetworkForAdvancedSGZone(DataCenter)},
+     * depending on the zone network type and whether or not security groups are enabled in the zone.
+     * @param dc - The zone (DataCenter) of the secondary storage VM.
+     * @return The default network for use with the secondary storage VM.
+     */
+    protected NetworkVO getDefaultNetworkForCreation(DataCenter dc) {
```

Comment:
Keeping the methods separate is actually very much on purpose. This increases the ability to extend and change core classes as needed. CS has a lot of code that is not very modular (1,000+ long line methods etc), and this change is for an increase in modularity. There is maybe some argument to be made for reducing code duplication though. The methods could be joined together in a secondary class used by both ConsoleProxyManagerImpl and SecondaryStorageManagerImpl. Some kind of SystemVmNetworkManager or something?